src/ELMduino.cpp

```diff
@@ -459,15 +469,29 @@ float ELM327::conditionResponse(const uint8_t &numExpectedBytes, const float &sc
     {
         if (debugMode)
             Serial.println("Lagging zeros found");
-
-        return ((float)(response >> (4 * payCharDiff)) * scaleFactor) + bias;
+        if (scaleFactor == 1 && bias == 0) // No scale/bias needed
+        {
+            return (response >> (4 * payCharDiff));
+        }
+        else 
+        {
+            return ((float)(response >> (4 * payCharDiff)) * scaleFactor) + bias;

```

Comment:
I think with your updates this line should have the typecast as a double

Reply:
Yeah, I think you’re right. Or just remove the typecast as the return value is implicitly cast to double anyway.